### PR TITLE
run rustfmt on redox

### PR DESCRIPTION
**Pull request by polymetric1**
I do not expect this PR to be pulled as-is (if at all). It should be reviewed for any unwanted formatting issues before being merged. Please don't hesitate to mark even a very minor change. it could be important. If you're worried that I somehow snuck some malicious code into this very hard to read diff, feel free to re-format everything and check the diff between my changes and yours.

### CODE REVIEW

**Comment by polymetric1** [comment]
This is a super big PR, I'm going to close and reopen multiple smaller ones

File: filesystem/apps/ansi-term/buffer.rs [hunk 1]
```diff
@@ -73,39 +73,39 @@ impl Buffer {
...
-            },
+            }
             Action::GotoColumn(x) => self.cursor.goto_x(x),
             Action::Goto(x, y) => self.cursor.goto(x, y),
             Action::GotoStart => self.cursor.goto(0, 0),
             Action::EraseAfter => {
                 self.grid[self.cursor.y].take(self.cursor.x..);
                 self.grid.take(self.cursor.y + 1..);
-            },
+            }
```
Comment (Ticki) [inline filesystem/apps/ansi-term/buffer.rs]: Hmm... Doesn't rustfmt do trailing commas?

Comment (polymetric1) [inline filesystem/apps/ansi-term/buffer.rs]: I don't know. Sometimes it adds them, sometimes it removes them. I can't tell where it does and doesn't do that though, it makes no sense.

Comment (Ticki) [inline filesystem/apps/ansi-term/buffer.rs]: Do rustfmt have settings?

Comment (polymetric1) [inline filesystem/apps/ansi-term/buffer.rs]: AFAIK the only formatting settings are `--write-mode` and some `--config-help` thing neither of which I've ever messed with

**Pull request closed without merging**